pkg/scheduler/ugm/user_tracker_test.go

```diff
@@ -171,55 +189,115 @@ func TestDecreaseTrackedResource(t *testing.T) {
 		t.Errorf("new resource create returned error or wrong resource: error %t, res %v", err, usage3)
 	}
 
-	removeQT, decreased = userTracker.decreaseTrackedResource(hierarchy1, TestApp1, usage4, true)
+	eventSystem.Reset()
+	removeQT, decreased = userTracker.decreaseTrackedResource(path1, TestApp1, usage4, true)
 	if !decreased {
-		t.Fatalf("unable to increase tracked resource: queuepath %+q, app %s, res %v, error %t", hierarchy1, TestApp1, usage1, err)
+		t.Fatalf("unable to increase tracked resource: queuepath %s, app %s, res %v, error %t", path1, TestApp1, usage1, err)
 	}
 	assert.Equal(t, 1, len(userTracker.getTrackedApplications()))
 	assert.Equal(t, removeQT, false, "wrong remove queue tracker value")
+	assert.Equal(t, 2, len(eventSystem.Events))
+	assert.Equal(t, si.EventRecord_UG_USER_RESOURCE, eventSystem.Events[0].EventChangeDetail)
+	assert.Equal(t, si.EventRecord_REMOVE, eventSystem.Events[0].EventChangeType)
+	assert.Equal(t, si.EventRecord_UG_APP_LINK, eventSystem.Events[1].EventChangeDetail)
+	assert.Equal(t, si.EventRecord_REMOVE, eventSystem.Events[1].EventChangeType)
 
 	usage5, err := resources.NewResourceFromConf(map[string]string{"mem": "10M", "vcore": "10"})
 	if err != nil {
 		t.Errorf("new resource create returned error or wrong resource: error %t, res %v", err, usage5)
 	}
-	removeQT, decreased = userTracker.decreaseTrackedResource(hierarchy2, TestApp2, usage5, true)
+	removeQT, decreased = userTracker.decreaseTrackedResource(path2, TestApp2, usage5, true)
 	if !decreased {
-		t.Fatalf("unable to increase tracked resource: queuepath %+q, app %s, res %v, error %t", hierarchy2, TestApp2, usage2, err)
+		t.Fatalf("unable to increase tracked resource: queuepath %s, app %s, res %v, error %t", path2, TestApp2, usage2, err)
 	}
 	assert.Equal(t, 0, len(userTracker.getTrackedApplications()))
 	assert.Equal(t, removeQT, true, "wrong remove queue tracker value")
 }
 
-func TestSetMaxLimits(t *testing.T) {
+func TestSetAndClearMaxLimits(t *testing.T) {
```

Comment:
Note: original unit test did not properly verify the behavior of the tracker, it only checked the return value of the increment which is always true (this should be changed in a follow-up JIRA). Now both canRunApp & headroom is validated with different settings.